plugins/techdocs-react/src/context.tsx

```diff
@@ -131,6 +127,14 @@ export const TechDocsReaderPageProvider = memo(
       defaultTechDocsReaderPageValue.shadowRoot,
     );
 
+    const metadata = useAsync(
+      () =>
+        shadowRoot
+          ? techdocsApi.getTechDocsMetadata(entityRef)
+          : new Promise<TechDocsMetadata>(() => {}),
+      [entityRef, shadowRoot],
```

Comment:
By waiting to fetch the metadata until the shadow root exists, the header is rendered in a loading state, as opposed to not rendering at all.

The downside of this is that it will wait longer than necessary to fetch the metadata for already generated pages (most cases). 

The alternative (current and also with other open PRs) is to assume that it exists, and reload when we know it exists, making too many calls in most cases.

I think this is more correct, but might appear slower in most cases.

Reply:
I haven't tested it; just an idea:
What if you use `useAsyncRetry` instead of `useAsync`? You can then have a separate `useEffect` that calls the metadata retry function if the metadata value is empty after the shadow root is loaded. Additionally, in the header component, you can render the loading state always when the metadata is loading or empty.

Reply:
@camilaibs It works, the only difference is that the header is not showing while the documentation is generating, but that's probably ok! Thanks for the suggestion.